genie-agent/src/main/java/com/netflix/genie/agent/execution/services/impl/ServicesAutoConfiguration.java

```diff
@@ -107,4 +110,27 @@ public KillService killService(final ApplicationEventPublisher applicationEventP
     public LaunchJobService launchJobService() {
         return new LaunchJobServiceImpl();
     }
+
+    /**
```

Comment:
Thought about it. I think this strategy makes sense i.e if the S3 credentials are not available then we provide a min functionality which can work without the credentials. In the case of archival that translates to "No archival". For `DownloadService` it can make sense to fail because it can just not function without the credentials. Hence I dont see a discrepancy if the guiding principle is to provide min functionality without the credentails

Reply:
I wouldn't call an archival service that silently *does not archive* "minimum functionality...

Reply:
If a user does not specify `archiveLocationPrefix` and does not care about archival then failing at start up does not make sense. Having a `NoOp` archival makes sense. On the other hand, if the user specifies `archiveLocationPrefix` and wants archival but does not provide S3 credentials then one should fail instead of not archiving. I have created a task to decide how we want to handle such cases. For the time being sticking to the `NoOp` implementation approach.